EIPS/eip-7870.md

```diff
@@ -0,0 +1,146 @@
+
+## Specification
+
+### Roles and Their Recommended Specifications
+
+Node operators typically run both an **Execution Layer (EL)** client and a **Consensus Layer (CL)** client on the same machine. The specifications below assume the combined resource usage of both.
+
+| Node Type               | Storage       | RAM   | CPU Cores / Threads | **PassMark CPU Rating**                            | Download / Upload speed  |
+|-------------------------|---------------|-------|---------------------|----------------------------------------------------|--------------------------|
+| **Full Node**           | 4 TB NVMe     | 32 GB | 4c / 8t             | ~1000 ST / 3000 MT                                 | 50 Mbps / 15 Mbps        |
+| **Attester**            | 4 TB NVMe     | 64 GB | 8c / 16t            | ~3500 ST / 25000 MT                                | 50 Mbps / 25 Mbps        |
+| **Local Block Builder** | 4 TB NVMe     | 64 GB | 8c / 16t            | ~3500 ST / 25000 MT                                | 100 Mbps / 50 Mbps       |
+
+*Approximate single-thread (ST) and multi-thread (MT) PassMark CPU scores. For example, a PassMark ST rating of 3500 and an MT rating of 25000 typically corresponds to upper mid-range server CPUs circa 2024–2025.*
+
+## Rationale
+
+### Storage
+
+- **Recommended**: 4 TB NVMe M.2 drive with:
+  - **Sequential R/W**: 7,000 MB/s
+  - **Random 4K R/W**: Up to 1,000,000 IOPS
+- **Why NVMe over SATA?**
+  - NVMe drives have significantly higher throughput and lower latency than SATA SSDs.  
+  - Drives without DRAM (DRAMless) or with QLC flash are not advised, due to lower endurance and potentially lower sustained performance.
+- **On Endurance (TBW)**  
+  - Running a node involves frequent writes (e.g., database updates, logs). Ensure that the SSD’s Total Bytes Written (TBW) rating is sufficient for multi-year operation.
+- **Capacity Considerations**  
+  - As of January 2025, 2 TB can still work, but daily chain growth makes 4 TB more future-proof.  
+  - While EIP-4444 aims to reduce historical storage requirements, the timeline for EIP-4444 remains uncertain.
+
+### Memory
+
+- **Why 64 GB for Validators (Attester / Local Block Builder)?**
+  - Running an Ethereum validator (both EL & CL clients) can be memory-intensive, with state cache dominating RAM usage.
+  - Certain memory intensive client combinations have historically failed with 16 GB.  
+  - Preliminary benchmarks (e.g., [zk-STARK memory usage](https://hackmd.io/@han/bench-hash-in-snark)) suggest cryptographic operations can demand significant RAM.  
+  - Relative to total hardware costs, upgrading from 32 GB to 64 GB is a small price but can improve stability and is more future proof with regards to zk-STARKS.
+
+### CPU
+
+- **Single vs. Multi-thread Performance**
+  - Attesters and local block builders benefit from both high single-thread performance and solid multi-thread performance.  
```

Comment:
The word "solid" stands out here. Feels too informal. I'd probably rephrase somehow. Maybe:
```suggestion
  - Attesters and local block builders benefit from both high single-thread and multi-thread performance.  
```
Also, this statement doesn't really compare ("single vs multi") or tell why it's important. Just that it is.